src/test/java/uk/gov/justice/hmpps/prison/repository/OffenderAlertRepositoryTest.java

```diff
@@ -126,4 +129,115 @@ void canRetrieveTheNameOfTheLastUserThatModifiedTheAlert() {
             .isNull();
 
     }
+
+    @Test
+    @DisplayName("can filter by offenderNo")
+    void canFilterByOffenderNo() {
+        final var filter = OffenderAlertFilter.builder().offenderNo("A1179MT").build();
+        final var alerts = repository.findAll(filter);
+
+        assertThat(alerts).hasSize(15);
+    }
+
+    @Test
+    @DisplayName("can filter by offenderNo but with just latest booking")
+    void canFilterByOffenderNoButWithJustLatestBooking() {
+        final var filter = OffenderAlertFilter.builder().offenderNo("A1179MT").latestBooking(true).build();
+        final var alerts = repository.findAll(filter);
+
+        assertThat(alerts).hasSize(3);
+    }
+
+    @Test
+    @DisplayName("can filter by alert codes")
+    void canFilterByAlertCodes() {
+        final var filter = OffenderAlertFilter
+            .builder()
+            .offenderNo("A1179MT")
+            .alertCodes("VOP,XCU")
+            .latestBooking(true)
+            .build();
+        final var alerts = repository.findAll(filter);
+
+        assertThat(alerts)
+            .hasSize(2)
+            .extracting(OffenderAlert::getCode)
+            .extracting(AlertCode::getCode)
+            .containsExactlyInAnyOrder("VOP", "XCU");
+
+    }
+
+    @Test
+    @DisplayName("can filter by code even when reference link is missing")
+    void canFilterByCodeEvenWhenReferenceLinkIsMissing() {
+        final var filter = OffenderAlertFilter
+            .builder()
+            .offenderNo("A1179MT")
+            .alertCodes("XTACT,XCU")
+            .latestBooking(true)
+            .build();
+        final var alerts = repository.findAll(filter);
+
+        assertThat(alerts)
+            .hasSize(2)
+            .extracting(OffenderAlert::getAlertCode)
+            .containsExactlyInAnyOrder("XTACT", "XCU");
+    }
+
+    @Test
+    @DisplayName("alert code description will be missing when reference data link missing")
```

Comment:
Bit confused by this test - it doesn't actually check alert code description - which I thought would be the same as the code rather than missing.

Reply:
The code is inside the the OffenderAlert.code object, so all I can do assert the object is null. In the API object the description is flattened so it easier to understand. At this level it is a missing object.

Perhaps a name change would make sense "alert code reference data with description will be missing when reference data link missing" - I'll add to next PR